### PR TITLE
Add data url format to read the key file

### PR DESCRIPTION
**Pull request by nodece**
### Motivation

Add data url format to read the key file, same with the java client.

This closes #866.

### CODE REVIEW

File: oauth2/client_credentials_provider.go [hunk 1]
```diff
@@ -55,8 +56,17 @@ func (k *KeyFileProvider) GetClientCredentials() (*KeyFile, error) {
...
 		keyFile, err = ioutil.ReadFile(filename)
-	case strings.HasPrefix(k.KeyFile, "data://"):
+	case strings.HasPrefix(k.KeyFile, DATA):
 		keyFile = []byte(strings.TrimPrefix(k.KeyFile, DATA))
+	case strings.HasPrefix(k.KeyFile, "data:"):
+		url, err := newDataURL(k.KeyFile)
+		if err != nil {
+			return nil, err
+		}
+		if url.Mimetype != "application/json" {
```
Comment (tisonkun) [inline oauth2/client_credentials_provider.go:66]: Shall we relax the restriction here but let the following `json.Unmarshal` to throw error if users don't specify `Mimetype` but actually parse a valid JSON payload?

**Review by tisonkun — APPROVED** [review]
LGTM. One minor comment inline.

File: oauth2/client_credentials_provider.go (hunk 1 above)
Comment (nodece) [reply oauth2/client_credentials_provider.go:66]: Sounds good to me.

File: oauth2/data_url_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	url, err = newDataURL(rawURL)
+	require.NoError(t, err)
+	assert.Equal(t, "application/json", url.Mimetype)
+	assert.Equal(t, "test", string(url.Data))
+
+	rawURL = "data://test"
+	url, err = newDataURL(rawURL)
+	require.Nil(t, url)
+	assert.Error(t, err)
+	assert.EqualError(t, errDataURLInvalid, err.Error())
```
Comment (tisonkun) [inline oauth2/data_url_test.go:57]: I suspect this test case will fail now.

Comment (nodece) [reply oauth2/data_url_test.go:57]: This test works fine.

**Comment by tisonkun** [comment]
Merging...

**Pull request merged**